include/zephyr/bluetooth/gatt.h

```diff
@@ -165,21 +165,117 @@ typedef ssize_t (*bt_gatt_attr_write_func_t)(struct bt_conn *conn,
 					     const void *buf, uint16_t len,
 					     uint16_t offset, uint8_t flags);
 
-/** @brief GATT Attribute structure. */
+/** @brief GATT Attribute
+ *
+ *  This type primarily represents an ATT Attribute that may be
+ *  an entry in the local ATT database. The objects of this type
+ *  must be part of an array that forms a GATT service.
+ *
+ *  While the formed GATT service is registered with the local
+ *  GATT server, pointers to this type can typically be given to
+ *  GATT server APIs, like bt_gatt_notify().
+ *
+ *  @note This type is given as an argument to the
+ *  bt_gatt_discover() application callback, but it's not a
+ *  proper object of this type. The permissions, read() and
+ *  write() methods are not available, and it's unsound to pass
+ *  the pointer to GATT server APIs.
+ */
 struct bt_gatt_attr {
-	/** Attribute UUID */
+	/** @brief Attribute Type, aka. "UUID"
+	 *
+	 *  The Attribute Type determines the interface can be expected
+	 *  from the read() and write() methods and the possible
+	 *  permission configurations.
+	 *
+	 *  E.g. Attribute of type @ref BT_UUID_GATT_CPF will act as a
+	 *  GATT Characteristic Presentation Format descriptor as
+	 *  specified in Core Specification 3.G.3.3.3.5.
+	 *
+	 *  Anyone can define a new Attribute Type.
+	 */
 	const struct bt_uuid *uuid;
-	/** Attribute read callback */
+
+	/** @brief Attribute Value read method
+	 *
+	 *  This is the generic interface to read from the Attribute
+	 *  Value of this object in on-air format.
```

Comment:
Should we mention that any values above 8 bits needs to be little endian, possible with reference to the helper functions?  

Ditto for the `write` callback

Reply:
I don't think we should. A service could define attributes that are big-endian, so this is not a certainty or requirement.

Reply:
> I don't think we should. A service could define attributes that are big-endian, so this is not a certainty or requirement.

ATT mandatates that multi-octet fields shall be sent as little-endian:

> Multi-octet fields within the Attribute Protocol shall be sent least significant octet first (little-endian)

Reply:
> Multi-octet fields within the Attribute Protocol shall be sent least significant octet first (little-endian) with the exception of the Attribute Value field. **The endianness of the Attribute Value field is defined by a higher layer specification.**

Core 3.F.3.3.1

Reply:
Oh. Yeah, that makes sense. So the core spec defined values are always little endian, but the attribute values can be whatever. Thanks